File: csharp/Behavioral/18-Observer/Observer/HomeownerNotifier.cs
namespace Observer;

public class HomeownerNotifier(string ownerName) : IJobCardObserver
{
    public void Update(string eventName, JobCard jobCard)
    {
        Console.WriteLine(
            $"  📧 [{ownerName}] Notification: '{eventName}' for job '{jobCard.Title}' — Status: {jobCard.Status}"
        );
    }
}

File: typescript/Behavioral/18-Observer/Observer/HomeownerNotifier.ts
import type { JobCard } from "../Subject/JobCard";
import { IJobCardObserver } from "./IJobCardObserver";

export class HomeownerNotifier implements IJobCardObserver {
  constructor(private readonly ownerName: string) {}

  update(eventName: string, jobCard: JobCard): void {
    console.log(
      `  📧 [${this.ownerName}] Notification: '${eventName}' for job '${jobCard.title}' — Status: ${jobCard.status}`,
    );
  }
}
